test/utils/asserts.cpp

```diff
@@ -14,7 +14,7 @@ std::ostream& operator<<(std::ostream& os, ExecutionResult result)
 
     os << "result(";
     if (result.has_value)
-        os << result.value;
+        os << result.value.i64;
```

Comment:
Maybe better to output this always in hex, because for floats decimal is going to be completely useless.

Reply:
I added additional hex output, but it is not complete as the expected part is formatted by the code in GTest macro.
```
Expected: result 300
  Actual: result(400 [0x190]) (of type fizzy::ExecutionResult)
```
It might be wise to expand the macro manually and clean up the code. Then we should also know the type of the result (based on the type of the provided expected value). Leaving that for some other time.